src/main/java/com/openshift/client/utils/TarFileUtils.java

```diff
@@ -35,7 +35,7 @@ public static boolean hasGitFolder(InputStream inputStream) throws IOException {
 		try {
 			boolean gitFolderPresent = false;
 			tarInputStream = new TarArchiveInputStream(new GzipCompressorInputStream(inputStream));
-			for (TarArchiveEntry entry = null; (entry = tarInputStream.getNextTarEntry()) != null;) {
```

Comment:
@adietish  Technically it wont matter because its a declaration and the the next statement will either give it a value or throw I would guess.  There really isn't another path it could take as far as I can see.